chains/substrate/listener.go

```diff
@@ -184,6 +184,14 @@ func (l *Listener) handleEvents(evts Events) {
 			l.log.Trace("Received ProposalFailed event", "depositNonce", evt.DepositNonce)
 		}
 	}
+
+	for range evts.System_CodeUpdated {
```

Comment:
I would check for the length like the other cases, otherwise lg2m.

Reply:
@GregTheGreek by checking length, do you mean the `subscription` checks? 
If you mean that, CodeUpdated is a special event and is separate from the subscriptions.

Reply:
If there are multiple CodeUpdated events in a single block, we only need to update the meta data once. So here we should check if there is atleast 1 event, and only call `updateMetadata` once